cluster-autoscaler/cloudprovider/azure/azure_cache.go

```diff
@@ -366,42 +392,60 @@ func (m *azureCache) FindForInstance(instance *azureRef, vmType string) (cloudpr
 		return nil, err
 	}
 	inst := azureRef{Name: resourceID}
+
+	if unowned, err := m.isUnownedInstance(inst, ""); unowned || err != nil {
+		return nil, err
+	}
+
+	if invalid, err := m.isInvalidVMType(inst, vmType, vmsPoolSet); invalid || err != nil {
+		return nil, err
+	}
```

Comment:
Please either undo this refactoring, or replace it with separate helpers for checking vs. marking as unowned:
* Neither `isUnownedInstance` nor `isInvalidVMType` return errors, no reason for (bool,error) signature
* It is very unintuitive/unexpected for `is*` functions to have a side effect (marking instance as unowned in this case); making it conditional on special value of second parameter in `isUnownedInstance` makes it worse

Reply:
Removed